File: frontend/src/app/services/audit.service.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { ApiService } from './api.service';

export interface AuditLog {
    id?: number;
    action: string;
    details: string;
    timestamp: string;
    username: string;
}

@Injectable({
    providedIn: 'root'
})
export class AuditService {
    constructor(private api: ApiService) { }

    getAll(): Observable<AuditLog[]> {
        return this.api.get<AuditLog[]>('audit');
    }
}

File: src/main/java/org/anpfacturationbackend/service/AuditService.java
package org.anpfacturationbackend.service;

import org.anpfacturationbackend.entity.AuditLog;
import org.anpfacturationbackend.repository.AuditLogRepository;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.stereotype.Service;

@Service
public class AuditService {

    private final AuditLogRepository auditLogRepository;

    public AuditService(AuditLogRepository auditLogRepository) {
        this.auditLogRepository = auditLogRepository;
    }

    public void log(String action, String details) {
        String username = SecurityContextHolder.getContext().getAuthentication() != null
                ? SecurityContextHolder.getContext().getAuthentication().getName()
                : "SYSTEM";

        AuditLog log = new AuditLog(username, action, details);
        auditLogRepository.save(log);
    }

    public java.util.List<AuditLog> getAll() {
        return auditLogRepository.findAll(org.springframework.data.domain.Sort
                .by(org.springframework.data.domain.Sort.Direction.DESC, "timestamp"));
    }
}
